minvws/nl-kat-coordination
# Add more handling of external services responses in scheduler

### Changes

* Made sure that `raise_for_status` was also done when `get` requests were issued
* When requesting a resource from an external source (e.g. requesting an object from octopoes) we don't raise the 404 exception, but we return `None` for the resource.
* Added more exception handling around get methods of the httpx client in the services.
* Added more exception handling in the boefje and normalizer scheduler when encountering exceptions

### Issue link

Closes #3368

### QA notes

We're going to remove an object, then we're going to update a schedule such that a rescheduled task gets executed. It should then see that the object is removed and it should then disable the schedule.

1. `make reset`

2. go through onboarding, scan mispo.es

3. get id from the schedule

```console
curl 'http://localhost:8004/schedules?schedule_hash=1eed929c42596710401d3cf02a7505f8'

# or with jq
curl 'http://localhost:8004/schedules?schedule_hash=1eed929c42596710401d3cf02a7505f8' | jq .results[0].id
```
4. we're going to patch a schedule to make sure that the deadline has passed, so you'll need the id from the last step

```console
curl -X PATCH http://localhost:8004/schedules/{id-from-last-step} -H "Content-Type: application/json" -d '{"deadline_at": "2024-08-15T00:00:00Z"}'
```
5. Now we're going to remove the object. Go to http://localhost:8000/en/dev/objects/detail/?ooi_id=Hostname%7Cinternet%7Cmispo.es and delete it

6. check if the schedule has been disabled (`enabled: false`) and the scheduler didn't crash. Be sure to wait at least a minute.

```console
curl http://localhost:8004/schedules/{id-from-last-step}

# or with jq
curl http://localhost:8004/schedules/{id-from-last-step} | jq .enabled
```

---

### Code Checklist







---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.
